kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/queues/FinalizingQueue.java

```diff
@@ -56,37 +53,35 @@ public FinalizingQueue(final String logPrefix, final int partition) {
   }
 
   /**
-   * See {@link WriteOnlyFinalizingQueue#scheduleForFinalization(AsyncEvent)}
+   * See {@link WriteOnlyFinalizingQueue#addFinalizableEvent}
    */
   @Override
-  public void scheduleForFinalization(
-      final AsyncEvent processedEvent
+  public void addFinalizableEvent(
+      final AsyncEvent event
   ) {
-    if (processedEvent.partition() != this.partition) {
-      log.error("attempted to finalize an event for partition {} on queue for partition {}",
-          processedEvent.partition(),
+    if (event.partition() != this.partition) {
+      final String errorMsg = String.format(
+          "Attempted to finalize an event for partition %d on queue for partition %d",
+          event.partition(),
           this.partition
       );
-      throw new IllegalStateException(String.format(
-          "attempted to finalize an event for partition %d on queue for partition %d",
-          processedEvent.partition(),
-          this.partition
-      ));
+      log.error(errorMsg);
+      throw new IllegalStateException(errorMsg);
     }
-    // Transition to OUTPUT_READY to signal that the event is done with processing
-    // and is currently awaiting finalization by the StreamThread
-    processedEvent.transitionToToFinalize();
```

Comment:
Another change of note: made this into a "pure" queue which doesn't do any event manipulations and just holds onto events. These event transitions now take place in the AsyncThreadPool class